compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/Window.desktop.kt

```diff
@@ -386,11 +386,15 @@ fun Window(
             it.compositionLocalContext = compositionLocalContext
             it.exceptionHandler = windowExceptionHandlerFactory.exceptionHandler(it)
 
+            val wasDisplayable = it.isDisplayable
+
             update(it)
 
-            if (!it.isDisplayable) {
-                it.makeDisplayable()
-                it.contentPane.paint(it.graphics)
+            // If displaying for the first time, make sure we draw the first frame before making
+            // the window visible, to avoid showing the window background
+            if (it.isDisplayable && !wasDisplayable) {
```

Comment:
Let's extract common code with Dialog here?

Reply:
It's not just these few lines that are duplicated here, but almost the entire `Window` and `Dialog` functions. It's also not such a trivial refactoring. So I prefer to not touch it in this PR.